client/packages/common/src/ui/forms/JsonForms/useJsonForms.tsx

```diff
@@ -26,38 +31,42 @@ import {
   arrayTester,
   Array,
 } from './components';
-
-// Temporarily hard-coded examples until we connect to database
-import patient from './jsonTemp/patient_1.json';
-import schema from './jsonTemp/schema.json';
-import uiSchema from './jsonTemp/ui-schema.json';
-import { JsonFormsRendererRegistryEntry } from '@jsonforms/core';
+import { useDocument } from './api';
 
 export type JsonData = {
   [key: string]: string | number | boolean | null | unknown | JsonData;
 };
 
 interface JsonFormsComponentProps {
   data: JsonData;
+  jsonSchema: JsonSchema;
+  uiSchema: UISchemaElement;
   setData: (data: JsonData) => void;
+  setError: (error: string | false) => void;
   renderers: JsonFormsRendererRegistryEntry[];
 }
 
 const FormComponent = ({
   data,
+  jsonSchema,
+  uiSchema,
   setData,
+  setError,
   renderers,
 }: JsonFormsComponentProps) => {
   return (
     <JsonForms
-      schema={schema}
+      schema={jsonSchema}
       uischema={uiSchema}
       data={data}
       renderers={renderers}
       // cells={materialCells}
       onChange={({ errors, data }) => {
         setData(data);
-        if (errors && errors.length) console.warn('Errors: ', errors);
+        if (errors && errors.length) {
+          setError(String(errors));
```

Comment:
errors is an array (`ErrorObject[]`) - is `String(errors)` the best option here?
perhaps an `errors?.map(({message}) => message ?? '').join(', ')` or errors[0] or an error array formatter? 
or `JSON.stringify(errors)`?